plugins/modules/sysrc.py

```diff
@@ -122,14 +122,13 @@ def has_unknown_variable(self, out, err):
         return err.find("unknown variable") > 0 or out.find("unknown variable") > 0
 
     def exists(self):
-        # sysrc doesn't really use exit codes
-        (rc, out, err) = self.run_sysrc(self.name)
+        # sysrc doesn't really use exit codes. Read all variables.
+        (rc, out, err) = self.run_sysrc('-e', '-a')
+        conf = dict([i.split('=') for i in out.splitlines()])
         if self.value is None:
-            regex = "%s: " % re.escape(self.name)
+            return self.name in conf
         else:
-            regex = "%s: %s$" % (re.escape(self.name), re.escape(self.value))
-
-        return not self.has_unknown_variable(out, err) and re.match(regex, out) is not None
+            return self.name in conf and conf[self.name] == '"%s"' % self.value
```

Comment:
Not holding the PR back because of that, but I would argue that this method could have a better name: it is doing more than just checking whether the variable exists or not, it is also verifying whether the existing value is equal to the incoming value. Maybe `is_value_equal()` or something... (I don't like that name, could not think of anything better though).

If no better name is found, at least I would add a comment indicating that be the top of the function.

Reply:
Hi @russoz, thank you for the review!

Most methods manipulating the name and value have single-word names (exists, contains, present, absent). The functionality depends on the context. I wouldn't change it. Changing the name of one method would be misleading and changing all of them would be messy.

I added a comment to the method exit. We can add other comments later.